lib/truncato/truncato.rb

```diff
@@ -26,26 +26,14 @@ def self.truncate source, user_options={}
   private
 
   def self.truncate_html source, options
+    source = source.encoding == Encoding::UTF_8 ? source.unicode_normalize : source
```

Comment:
Note: Unicode normalization is supported not only on UTF-8: https://docs.ruby-lang.org/en/3.4/String.html#method-i-unicode_normalize

As of Ruby 3.4:

> The encoding of self must be one of:
> * Encoding::UTF_8
> * Encoding::UTF_16BE
> * Encoding::UTF_16LE
> * Encoding::UTF_32BE
> * Encoding::UTF_32LE
> * Encoding::GB18030
> * Encoding::UCS_2BE
> * Encoding::UCS_4BE